traitsui/testing/tester/qt4/helpers.py

```diff
@@ -105,11 +107,15 @@ def mouse_click_qwidget(control, delay):
     delay : int
         Time delay (in ms) in which click will be performed.
     """
-    QTest.mouseClick(
-        control,
-        QtCore.Qt.LeftButton,
-        delay=delay,
-    )
+    if isinstance(control, QtGui.QAbstractButton):
+        time.sleep(delay/1000)
```

Comment:
I'm not a huge fan of this `isinstance` check, or the use of time.sleep here.  At the time of writing this they felt like the best solution.

Reply:
`QTest.qSleep` would be better which is designed for this (you might need to check the delay is larger than 0).
The `isinstance(control, QtGui.QAbstractButton)` is okay because we are checking against an interface, not a concrete implementation. We might want to comment on why we are not using `mouseClick` directly: `mouseClick` assumes the center of the widget, which is not the appropriate default location when the widget is a button.